### ISSUE TITLE
Continuous Write to %TEMP%\Setup Log YYYY-MM-DD #001.txt 

### ISSUE THREAD

**Issue by MrJimmo** [microsoft/vscode#195194]
Type: <b>Performance Issue</b>

I noticed the file %TEMP%\Setup Log 2023-09-14 #001.txt  was being written to every second.

"Application is still running, waiting" was being written over and over.

VSCode had been running since 9/14 (Machine reboot last Patch Tuesday update), so this temp file was now 104MB.

This is wasteful drive writes and space usage. 

VS Code version: Code 1.82.2 (abd2f3db4bdb28f9e95536dfa84d8479f1eb312d, 2023-09-14T05:55:25.390Z)
OS version: Windows_NT x64 10.0.19045
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|13th Gen Intel(R) Core(TM) i9-13900KF (32 x 2995)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|127.78GB (84.40GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (13)</summary>

Extension|Author (truncated)|Version
---|---|---
csharp|ms-|2.5.30
vscode-dotnet-runtime|ms-|1.8.0
python|ms-|2023.18.0
vscode-pylance|ms-|2023.10.10
jupyter|ms-|2023.8.1002501831
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.0.17
vscode-jupyter-cell-tags|ms-|0.1.8
vscode-jupyter-slideshow|ms-|0.1.5
remote-wsl|ms-|0.81.7
cpptools|ms-|1.17.5
powershell|ms-|2023.6.0
platformio-ide|pla|3.3.1

</details>

**Comment by VSCodeTriageBot** [comment]
Thanks for creating this issue! It looks like you may be using an old version of VS Code, the latest stable release is 1.83.0. Please try upgrading to the latest version and checking whether this issue remains.

Happy Coding!

**Comment by DonJayamanne** [comment]
Please could you disable all extensions and verify you are still running into this issue.

**Comment by MrJimmo** [comment]
As was suggested, I upgraded and the writing to the file stopped, but I think that's expected as the log was just getting filled with "Application is still running, waiting"

My guess is that VSCode gets into this state when it's waiting to be restarted after/before taking an update or some other 'pending' type of operation, and just spews this log line over and over.


Here's snips of the log file, First line, and then 2 other sections snipped, and the last few lines of the log at the bottom....

2023-09-14 10:52:20.292   Log opened. (Time zone: UTC-07:00)
2023-09-14 10:52:20.292   Setup version: Inno Setup version 6.0.5 (u)
2023-09-14 10:52:20.292   Original Setup EXE: C:\Users\jimm\AppData\Local\Temp\vscode-stable-user-x64\CodeSetup-stable-abd2f3db4bdb28f9e95536dfa84d8479f1eb312d.exe
2023-09-14 10:52:20.292   Setup command line: /SL5="$F128C,93275017,828416,C:\Users\jimm\AppData\Local\Temp\vscode-stable-user-x64\CodeSetup-stable-abd2f3db4bdb28f9e95536dfa84d8479f1eb312d.exe" /verysilent /log /update="C:\Users\jimm\AppData\Local\Temp\vscode-stable-user-x64\CodeSetup-stable-abd2f3db4bdb28f9e95536dfa84d8479f1eb312d.flag" /nocloseapplications /mergetasks=runcode,!desktopicon,!quicklaunchicon
2023-09-14 10:52:20.292   Windows version: 10.0.19045  (NT platform: Yes)
2023-09-14 10:52:20.292   64-bit Windows: Yes
2023-09-14 10:52:20.292   Processor architecture: x64
2023-09-14 10:52:20.292   User privileges: None
2023-09-14 10:52:20.306   Administrative install mode: No
2023-09-14 10:52:20.306   Install mode root key: HKEY_CURRENT_USER
2023-09-14 10:52:20.306   64-bit install mode: Yes
2023-09-14 10:52:20.307   Created temporary directory: C:\Users\jimm\AppData\Local\Temp\is-BJQHM.tmp
2023-09-14 10:52:20.450   Starting the installation process.
2023-09-14 10:52:20.576   Directory for uninstall files: C:\Users\jimm\AppData\Local\Programs\Microsoft VS Code
2023-09-14 10:52:20.580   Will append to existing uninstall log: C:\Users\jimm\AppData\Local\Programs\Microsoft VS Code\unins000.dat
2023-09-14 10:52:20.580   -- File entry --
2023-09-14 10:52:20.580   Dest filename: C:\Users\jimm\AppData\Local\Programs\Microsoft VS Code\unins000.exe
2023-09-14 10:52:20.580   Non-default bitness: 32-bit
2023-09-14 10:52:20.581   Time stamp of our file: 2023-09-14 10:52:20.088
2023-09-14 10:52:20.581   Dest file exists.
[...snip lots of File and Registry entry output...]
2023-09-14 10:52:26.224   Deleting uninstall key left over from previous non administrative install.
2023-09-14 10:52:26.224   Creating new uninstall key: HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Uninstall\{771FD6B0-FA20-440A-A002-3B3BAC16DC50}_is1
2023-09-14 10:52:26.224   Writing uninstall key values.
2023-09-14 10:52:26.224   Detected previous administrative 64-bit install? No
2023-09-14 10:52:26.224   Detected previous administrative 32-bit install? No
2023-09-14 10:52:26.229   Installation process succeeded.
2023-09-14 10:52:26.229   Application is still running, waiting
2023-09-14 10:52:27.242   Application is still running, waiting
2023-09-14 10:52:28.247   Application is still running, waiting
[...25 days of this same "Application is still running, waiting" string, until I restart VSCode today...]
2023-10-09 14:30:27.224   Application is still running, waiting
2023-10-09 14:30:28.230   Application is still running, waiting
2023-10-09 14:30:29.242   Application is still running, waiting
2023-10-09 14:30:32.403   Need to restart Windows? No
2023-10-09 14:30:32.409   -- Run entry --
2023-10-09 14:30:32.409   Run as: Original user
2023-10-09 14:30:32.409   Type: Exec
2023-10-09 14:30:32.409   Filename: C:\Users\jimm\AppData\Local\Programs\Microsoft VS Code\Code.exe
2023-10-09 14:30:32.439   Deinitializing Setup.
2023-10-09 14:30:32.446   Log closed.


I'd say search VSCode code prior to 1.83.0 and search for "Application is still running, waiting" and that may provide some clues. I'm not seeing that string in latest on https://github.com/microsoft/vscode  ...but not sure public gets full history.

Also, looks like my issue is same as:  https://github.com/microsoft/vscode/issues/192486   which got closed.


-jimm

________________________________
